File: javascript/p2.js
#!/usr/bin/env node

let limit = 4000000;
let sum = 0;
let a = 1;
let b = 1;
let c = 2;
if(process.argv.length > 2){
  max = process.argv[2]
}
while(c < limit){
  sum += c;
  a = b + c;
  b = c + a;
  c = a + b;
}

console.log(sum);

File: c/p2.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{
  int limit = 4000000;
  int sum = 0;
  int a = 1;
  int b = 1;
  int c = 2;
  if (argc > 1) {
    limit = atoi(argv[1]);
  }
  while (c < limit) {
    sum += c;
    a = b + c;
    b = c + a;
    c = a + b;
  }
  printf("%d\n", sum);
}
